crates/builder/src/main.rs

```diff
@@ -1,102 +1,55 @@
-    pub confirmation_buffer: u64,
-    /// address of the Zenith contract
-    pub zenith: Address,
-    /// URL for Quincey server to sign blocks. This prop is disregarded if a
-    /// local_sequencer_signer is configured via the "SEQUENCER_KEY" env var.
-    pub quincey_url: Cow<'static, str>,
-    /// URL for RPC node
-    pub rpc_url: Cow<'static, str>,
-
-    /// Wallet for signing blocks locally.
-    pub local_sequencer_signer: Option<LocalOrAws>,
-
-    /// Whether to use calldata or blob for transactions
-    pub use_calldata: bool,
-}
-
-impl ChainConfig {
-    /// Creates a new ChainConfig for the Holesky testnet.
-    const fn holesky(
-        local_sequencer_signer: Option<LocalOrAws>,
-        rpc_url: Cow<'static, str>,
-        quincey_url: Cow<'static, str>,
-        zenith: Address,
-    ) -> Self {
-        ChainConfig {
-            host_chain_id: 17000,
-            ru_chain_id: 17001,
-            confirmation_buffer: 60 * 20,
-            zenith,
-            quincey_url,
-            rpc_url,
-            local_sequencer_signer,
-            use_calldata: true,
-        }
-    }
-}
-
 #[tokio::main]
 async fn main() -> eyre::Result<()> {
     tracing_subscriber::fmt::try_init().unwrap();
-
     let span = tracing::info_span!("zenith-builder");
 
-    // finish app config by loading key from env
-    let config = ChainConfig::holesky(None, load_rpc()?, load_quincey()?, load_zenith_address()?);
-
-    // Load builder key from env
-    let wallet = LocalOrAws::load("BUILDER_KEY_ID", Some(config.host_chain_id)).await?;
-    let builder_rewards = wallet.address();
+    // load config from env
+    let config = load_builder_config().await?;
 
+    // build provider from config
     let provider = ProviderBuilder::new()
         .with_recommended_fillers()
-        .signer(EthereumSigner::from(wallet))
-        .on_builtin(&config.rpc_url)
+        .signer(EthereumSigner::from(config.builder_signer.clone()))
+        .on_builtin(&config.host_rpc_url.clone())
         .await?;
-    tracing::debug!(rpc_url = config.rpc_url.as_ref(), "instantiated provider");
-    let zenith = Zenith::new(config.zenith, provider.clone());
 
-    let build = tasks::block::BlockBuilder { wait_secs: 5 };
+    tracing::debug!(
+        rpc_url = config.host_rpc_url.as_ref(),
+        "instantiated provider"
+    );
+
+    // build zenith from config
+    let zenith = Zenith::new(config.zenith_address, provider.clone());
+
+    let build = tasks::block::BlockBuilder::new(&config);
+
     let submit = tasks::submit::SubmitTask {
         provider,
         zenith,
         client: reqwest::Client::new(),
         config,
-        builder_rewards,
-        gas_limit: 30_000_000,
     };
 
     let (submit_channel, submit_jh) = submit.spawn();
 
     let (build_channel, build_jh) = build.spawn(submit_channel);
 
+    // server
     let server = serve_builder_with_span(build_channel, ([0, 0, 0, 0], 6969), span);
```

Comment:
should the port be a configurable variable?

Reply:
yes pls

Reply:
https://github.com/init4tech/zenith-rs/pull/12/commits/c868567ab3668f61ecab9819374aed71876ed2f1